ibis/bigquery/tests/test_client.py

```diff
@@ -375,3 +369,31 @@ def test_scalar_param_partition_time(parted_alltypes):
     expr = t[t.PARTITIONTIME < param]
     df = expr.execute(params={param: '2017-01-01'})
     assert df.empty
+
+
+def test_exists_table(client):
+    assert client.exists_table('functional_alltypes')
+    assert not client.exists_table('footable')
+
+
+def test_exists_database(client):
+    assert client.exists_database('testing')
+    assert not client.exists_database('foodataset')
+
+
+@pytest.mark.parametrize('kind', ['date', 'timestamp'])
+@pytest.mark.parametrize(
+    ('option', 'expected_fn'),
+    [
+        (None, 'my_{}_parted_col'.format),
+        ('PARTITIONTIME', lambda kind: 'PARTITIONTIME'),
+        ('foo_bar', lambda kind: 'foo_bar'),
+    ]
+)
+def test_parted_column(client, kind, option, expected_fn):
+    table_name = '{}_column_parted'.format(kind)
+    option_key = 'bigquery.partition_col'
+    with ibis.config.option_context(option_key, option):
```

Comment:
Why does Ibis have an option to rename the partition column? Or does this do something else?

Reply:
This option is a convenience to allow users to refer to `_PARTITIONTIME`, since refering to it without renaming generates an error in BigQuery. For column-based partitioning this isn't a problem, but for tables that are implicitly partitioned we want to do this.